src/__tests__/Document.test.ts

```diff
@@ -856,7 +856,46 @@ describe('validate', () => {
 			});
 			const document = new DocumentSubclass(schema, { data: { prop1: 'foo' } });
 
-			const expected = new Map([['prop1', 'Test error message']]);
+			const expected = new Map([['prop1', ['Test error message']]]);
+			expect(await document.validate()).toEqual(expected);
+		});
+
+		test('should return error at property & unravel nested errors if schemaType validation fails', async () => {
+			const nestedSchema: SchemaDefinition = {
+				nestedProp1: [{ type: 'number', path: '1', required: true }],
+				nestedProp2: { type: 'number', path: '2', required: true },
+			};
+
+			const subDocumentArraySchema: SchemaDefinition = {
+				prop1: [{ type: 'string', path: '7', required: true }],
+				prop2: [{ type: 'number', path: '8', dbDecimals: 2, required: true }],
+			};
+
+			const definition: SchemaDefinition = {
+				prop1: { type: 'string', path: '1' },
+				prop2: { type: 'number', path: '2', dbDecimals: 2, required: true },
+				array: [{ type: 'string', path: '3' }],
+				nestedArray: [[{ type: 'string', path: '4', required: true }]],
+				nestedObject: {
+					prop1: { type: 'string', path: '5' },
+					prop2: { type: 'number', path: '6', dbDecimals: 2, required: true },
+				},
+				subdocumentArray: [subDocumentArraySchema],
+				embeddedType: nestedSchema,
+			};
+			const schema = new Schema(definition);
+			const document = new DocumentSubclass(schema, {
+				record: ['55', null, ['foo', 'bar'], [null], 'bing', null, [[null, '5']], [['5', null]]],
+			});
+
+			const expected = new Map([
```

Comment:
I think this would be a better exercise if you had "good" and "bad" variants of each type. For instance, there's no exercising of the regular array here because its passing. This test therefore doesn't confirm that arrays are working as expected.

So I think you can make have each data structure represented 2x, one that should pass and one that should fail. You could prefix each with `passing` and `failing` or something like that.